src/Components/Legend.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Segment, Sidebar, Grid, Menu, Sticky, Icon, Header } from 'semantic-ui-react'
+
+class Legend extends Component {
+
+  hideLegend = async () => {
+    await this.props.dispatch({ type: 'CHANGE_LEGEND_VISIBILITY' })    
+  }
+
+  render() {
+    return (
+      <>
+        <Sidebar
+          className='legend-content'
+          id='legend'
+          as={Segment}
+          animation='overlay'
+          direction='bottom'
+          visible={this.props.visible}
+          onClick={this.hideLegend}
+        >
+          <Grid >
+            <Grid.Column textAlign='center' className="legend-padding" >
+              <Icon
+                name='window minimize outline' 
+                color='grey' 
```

Comment:
No need for `-`between words in `name=`?

Reply:
No, since its the name of the icon

Reply:
Noticed I was a bit unclear, Semantic's names for icons are without ```-```